### PR TITLE
Render destructured struct function param names as underscore

### LINKED ISSUE #83852: Invalid rendering for function parameters using destructuring
This code:

```rust
pub fn body_owner(&self, BodyId { hir_id }: BodyId) -> HirId {
   // ...
}
```

renders the same in rustdoc, whereas it should be rendered like this:

```rust
pub fn body_owner(&self, _: BodyId) -> HirId;
```

### PR DESCRIPTION
**Pull request by eggyal**
Fixes #83852

r? @GuillaumeGomez

### CODE REVIEW

File: src/test/rustdoc/issue-83852.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#![crate_name = "foo"]
```
Comment (jyn514) [inline src/test/rustdoc/issue-83852.rs:1]: Can you rename this file to something more descriptive, like `struct-arg-pattern.rs`?

**Review by jyn514 — COMMENTED** [review]
r=me with the nit addressed - thanks for tackling this (and so quickly!)

**Comment by jyn514** [comment]
@bors r+ squash

FYI you can avoid commits like "Merge branch 'master' of github.com:rust-lang/rust into issue-83852" by rebasing: https://rustc-dev-guide.rust-lang.org/git.html#rebasing

**Comment by eggyal** [comment]
Thanks... I was just about to push a rebased and squashed commit... wasn't paying close enough attention!

**Comment by jyn514** [comment]
Haha no worries :) feel free to push the rebased commit if you want bors to show your name on the commit message.

**Comment by eggyal** [comment]
I think I'll survive 😂

By the way, when you said `r=me` earlier, is that just a signal to other reviewers that you're happy for the PR to be marked approved by you or should I have then `@bors r+`d after the nit was resolved (had assumed I don't have permission for that).

**Comment by jyn514** [comment]
> By the way, when you said r=me earlier, is that just a signal to other reviewers that you're happy for the PR to be marked approved by you

Yes, exactly - and it also reminds me later that I should remember to say r+ :laughing:

>  or should I have then @bors r+d after the nit was resolved (had assumed I don't have permission for that).

Correct, you don't have permissions unless you're on one of the rust-lang teams.

**Comment by jyn514** [comment]
Also it tells you that the PR is almost done and I'm only picking on small things because everything else is working :)

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
That was fast, thanks a lot! :)

**Pull request merged**